js/common/lib/training-session.ts

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {InferenceSession} from './inference-session.js';
+import {TrainingSession as TrainingSessionImpl} from './training-session-impl.js';
+
+/* eslint-disable @typescript-eslint/no-redeclare */
+
+/**
+ * Represent a runtime instance of an ONNX training session,
+ * which contains a model that can be trained, and, optionally,
+ * an eval and optimizer model.
+ */
+export interface TrainingSession {
+  // #region run()
+
+  // /**
+  //  * Run TrainStep asynchronously with the given feeds and options.
+  //  *
+  //  * @param feeds - Representation of the model input. See type description of `InferenceSession.InputType` for
+  //  detail.
+  //  * @param options - Optional. A set of options that controls the behavior of model training.
+  //  * @returns A promise that resolves to an error code
+  //  */
+  // runTrainStep(feeds: InferenceSession.FeedsType, options?: InferenceSession.RunOptions):
+  // Promise<InferenceSession.ReturnType>;
+
```

Comment:
they are all commented... did you forgot push some local changes?

Reply:
ah, no, when the implementation threw an error, it was causing a lot of linting errors so I just pushed them as commented so that people can see what the functions will look like + some of the functionality of TrainingSession